docs/webactions.md

```diff
@@ -0,0 +1,133 @@
+             code: 200,
+             body: png };
+}
+```
+
+It is important to be aware of the [response size limit](reference.md) for actions since a response that exceeds the predefined system limits will fail. Large objects should not be sent inline through OpenWhisk, but instead deferred to an object store, for example.
+
+## Handling HTTP requests with actions
+
+An OpenWhisk action that is not a web action requires both authentication and must respond with a JSON object. In contrast, web actions may be invoked without authentication, and may be used to implement HTTP handlers that respond with _headers_, _status code_, and _body_ content of different types. The web action must still return a JSON object, but the OpenWhisk system (namely the `controller`) will treat a web action differently if its result includes one or more of the following as top level JSON properties:
+
+1. `headers`: a JSON object where the keys are header-names and the values are string values for those headers (default is no headers).
+2. `code`: a valid HTTP status code (default is 200 OK).
+3. `body`: a string which is either plain text or a base64 encoded string (for binary data).
+
+The controller will pass along the action-specified headers, if any, to the HTTP client when terminating the request/response. Similarly the controller will respond with the given status code when present. Lastly, the body is passed along as the body of the response. Unless a `content-type header` is declared in the action result’s `headers`, the body is passed along as is if it’s a string (or results in an error otherwise). When the `content-type` is defined, the controller will determine if the response is binary data or plain text and decode the string using a base64 decoder as needed. Should the body fail to decoded correctly, an error is returned to the caller.
+
+## HTTP Context
+
+A web action, when invoked, receives all the HTTP request information available as additional parameters to the action input argument. They are:
+
+1. `__ow_meta_verb`: the HTTP method of the request.
+2. `__ow_meta_headers`: the request headers.
+3. `__ow_meta_path`: the unmatched path of the request (matching stops after consuming the action extension).
+
+The request may not override any of the named `__ow_` parameters above; doing so will result in a failed request with status equal to 400 Bad Request.
+
+## Additional features
+
+Web actions bring some additional features that include:
+
+1. `Content extensions`: the request must specify its desired content type as one of `.json`, `.html`, `.text` or `.http`. This is done by adding an extension to the action name in the URI, so that an action `/guest/demo/hello` is referenced as `/guest/demo/hello.http` for example to receive an HTTP response back.
+2. `Projecting fields from the result`: the path that follows the action name is used to project out one or more levels of the response. For example, 
```

Comment:
well it is true - if you follow the contract.
enhancing/changing this is subject of the issue you opened.